doc/design/design_model_zoo.md

```diff
@@ -0,0 +1,227 @@
+   ```
+1. [Transfer learning](https://en.wikipedia.org/wiki/Transfer_learning) to fit a new dataset with the same features:
+   ```sql
+   SELECT * FROM iris.transfer_iris_samples
+   TRAIN sqlflow.org/modelzoo/iris_dnn_128x32/v1.0
+   USING sqlflow.org/modelzoo/iris_dnn_128x32/v1.0
+   INTO modeldb.my_iris_dnn_model;
+   ```
+
+## Concept of A Model in the Model Zoo
+
+A Model in the Model Zoo contains below components:
+
+1. The model definition. A python program defines the model's structure, currently it can be:
+    - TensorFlow Estimator
+    - A Keras subclass model
+    - A XGBoost model type
+1. Pre-trained model parameters.
+1. Model publication information, including:
+    - Unique model name
+    - Author
+    - Version
+    - Hyper Parameters used to train the model parameters
+    - How to convert database input to model input
+
+Note that the model definition may have configurable hyper-parameters for training, like the size
+and depth of the DNN layers, loss function and optimizer used to train the model. We can train
+many different model parameters with different hyper parameter settings use one model definition.
+Also we can train the model parameters using different dataset. So the model definition can be reused
+to form many "models" to solve the real-world problem, but the dataset, hyper-parameters, model parameters
+are unique for each "model" to solve a problem.
+
+The "model" in the model zoo does not mean a model definition here, but a "model" with model parameters
+and other information used to solve one real-world problem. So we always use the dataset name, model name,
+model structure description to form a "model name" to indicate that this model is used to solve this kind
+of problem.
+
+## The Model Zoo Hosting Service
+
+The model zoo hosting service is a file service that can be accessed from the internet.
+It serves all available models and corresponding pre-trained weight. We may need to use
+a [CDN](https://en.wikipedia.org/wiki/Content_delivery_network) service if the traffic
+becomes large.
+
+The publication of a trained model needs to upload the parameters of the model to a
+storage service, labeled with the version, or Git commit ID, of the model zoo.
+The model directory is of the format: `sqlflow.org/modelzoo/iris_dnn_128x32/v1.0/`.
+`iris_dnn_128x32` is the model's name, and `v1.0` is the version.
+The directory name is responsible to explain the model's type, network structure and which dataset is
+used to train the pre-trained parameters. You can access `sqlflow.org/modelzoo/v1.0/README.md`
+from the browser to get the model's full documentation. All models under `sqlflow.org/modelzoo` are
+developed under `https://github.com/sql-machine-learning/models` parameters is only stored under
+`sqlflow.org` but not under Github.
+
+The content of the directory should be like:
+
+```
+iris_dnn_128x32/v1.0/
+   - model_meta.json  # model information useful for load and run.
```

Comment:
I would suggest using the `.ymal` file, which carries comments so as to more readable.

Reply:
Sounds good, will update in next PR